Packs/Celonis/Integrations/CelonisEventCollector/CelonisEventCollector.py

```diff
@@ -0,0 +1,240 @@
+        Returns:
+            dict: The audit logs in JSON format.
+        """
+        results = self._http_request(
+            method="GET",
+            url_suffix=f"/log/api/external/audit?pageNumber={PAGE_NUMBER}&pageSize={PAGE_SIZE}&from={start_date}&to={end_date}",
+            headers={
+                'Authorization': f'Bearer {self.token}',
+            },
+            resp_type='response',
+            retries=3
+        )
+        return results
+
+
+""" HELPER FUNCTIONS """
+
+
+def sort_events_by_timestamp(events: list) -> list:
+    """
+    Sorts a list of events by their date in ascending order.
+    Args:
+        events (list): A list of dictionaries.
+    Returns:
+        list: The sorted list of events based on the 'timestamp' field.
+    """
+    return sorted(events, key=lambda x: datetime.strptime(x['timestamp'], '%Y-%m-%dT%H:%M:%S.%f%z'))
+
+
+def add_millisecond(timestamp: str) -> str:
+    """
+    Adds one millisecond to a given timestamp.
+    Args:
+        timestamp (str): The timestamp in ISO 8601 format (e.g., "2025-02-05T14:30:00.123Z").
+    Returns:
+        str: The new timestamp with one millisecond added, formatted in ISO 8601.
+    """
+    dt = datetime.strptime(timestamp, DATE_FORMAT)
+    dt += timedelta(milliseconds=1)
+    return dt.strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3] + "Z"
+
+
+""" COMMAND FUNCTIONS """
+
+
+def test_module(client: Client) -> str:
+    """
+    Tests the connection to the service by creating an access token.
+    Args:
+        client (Client): The client object used to interact with the service.
+    Returns:
+        str: 'ok' if the connection is successful. If an authorization error occurs, an appropriate error message is returned.
+    """
+    current_time = get_current_time()
+    start_date = (current_time - timedelta(minutes=1)).strftime(DATE_FORMAT)
+    end_date = current_time.strftime(DATE_FORMAT)
+    fetch_events(client, 1, {'start_date': start_date, 'end_date': end_date})
+    return "ok"
+
+
+def fetch_events(client: Client, fetch_limit: int, get_events_args: dict = None) -> tuple[list, dict]:
+    if get_events_args:  # handle get_event command
+        start_time = get_events_args.get('start_date', '')
+        end_time = get_events_args.get('end_date', '')
+    else:  # handle fetch_events case
+        last_run = demisto.getLastRun() or {}
+        start_time = last_run.get('start_date', '')
+        client.set_token(last_run.get('audit_token', ''))
+        if not start_time:
+            start_time = get_current_time().strftime(DATE_FORMAT)
+        end_time = get_current_time().strftime(DATE_FORMAT)
+
+    demisto.debug(f'Fetching audit logs events from date={start_time} to date={end_time}.')
+
+    output: list = []
+    while True:
+        try:
+            response = client.get_audit_logs(start_time, end_time)
+        except Exception as e:
+            if e.res.status_code == 429:
```

Comment:
Why do we stop the process when we get the rate limit? you should wait and retry instead of stopping. 
something like:
```
        if hasattr(e, "res") and hasattr(e.res, "status_code"):
            if e.res.status_code == 429:
                retry_after = int(e.res.headers.get('x-ratelimit-reset', 5)) 
                demisto.debug(f"Rate limit reached. Waiting {retry_after} seconds before retrying.")
                time.sleep(retry_after)
                continue  # Retry after wait
```

Reply:
No need to handle the rate-limiting twice. If the rate-limit happens it will handle it in the exception and it will avoid unnecessary sleeping